### ISSUE TITLE
Elspotpris.dk bruger for meget cpu (selvom der ikke er interaktion med siden)

### ISSUE THREAD

**Issue by bennybechdk** [rndfm/elspotpris#71]
På min maskine bruger min chrome med ca 15 forskellige faner åben ca 5% cpu (ifølge joblisten) , åbner jeg elsportpris.dk stiger stiger det til 3%
Tilsvarende hvis hop i cpu forbrug sker  hvis jeg gør det i Firefox.

**Comment by rndfm** [comment]
Jeg har også oplevet at siden for nyligt er blevet langsommere på enheder med lille cpu.

**Comment by rndfm** [comment]
Jeg har kigget en smule ind i det. 
Det så ud til at det lå ud over koden fra elspotpris.dk og mere var en svelte ting.

Jeg har prøvet at opgraderer svelte til en nyere version og jeg synes at min tablet med meget begrænset CPU igen kan være med.

Kan du prøve hvad du oplever nu?

**Comment by rndfm** [comment]
Og lige et spørgsmål mere. Hvad du graf eller tabel visningen åben? Og har du forsøgt om det gør en forskel?

**Comment by rndfm** [comment]
Jeg fandt lige en performance fælde mere.

**Comment by bennybechdk** [comment]
Det er stort set kun ved graf:
![image](https://user-images.githubusercontent.com/14022402/198397392-d61e574f-f549-46c5-8f20-a2fbbf1e8d75.png)

PS.Kan se at jeg fik skrevet 3% mente knap 30% (beklager den dårlige korrektur)

**Comment by rndfm** [comment]
Jeg har på fornemmelsen at det er co2 visningen

**Comment by rndfm** [comment]
Min langsigtede plan er ikke at bruge et tredigepartskomponent (bruger pt apexcharts) og i stedet bygge en letvægtsløsning specifikt til elspotpris.dk.

Lige nu og her kan vi sætte antallet af punkter ned for CO2 og CO2 prognosen. Det er mit bedste bud på at sætte forbruget lidt ned.

**Comment by hekner** [comment]
Der er også et tidsaspekt. Hvis jeg har en tab (Chrome Windows) åben på siden i lang tid, bliver den langsommere og langsommere - til sidst helt ubrugelig (kan tage 30 sek fra jeg trykker på en fluebens box til den skifter). Hvis jeg så lukker tab'en og åbner en ny, virker det hele fint igen - en dags tid. Så bliver det langsomt igen.

**Comment by rndfm** [comment]
> Der er også et tidsaspekt. Hvis jeg har en tab (Chrome Windows) åben på siden i lang tid, bliver den langsommere og langsommere - til sidst helt ubrugelig (kan tage 30 sek fra jeg trykker på en fluebens box til den skifter). Hvis jeg så lukker tab'en og åbner en ny, virker det hele fint igen - en dags tid. Så bliver det langsomt igen.

Er det også kun graf visningen der bliver langsommere?

**Comment by hekner** [comment]
Ja, det er kun i graf mode. Her ser TaskMgr således ud når tingene er sløvet ned:

![image](https://user-images.githubusercontent.com/48845202/200107003-b3468bb9-7ad1-4976-a59d-65ac1ac3e2f4.png)

Hvis jeg så skifter til tabel mode, falder task CPU med det samme til ~0%, og alt er responsivt igen. Memory forbruget er uændret. Hvis jeg så skifter til graf mode igen, bliver den u-responsiv med det samme. Det kan kun repareres ved at lukke tab'en og åbne en ny.

**Comment by ogregpt** [comment]
Ved ikke om dette er beslægtet eller jeg skal starte nyt issue:
Glasswire netmonitor viser at firefox hos elspotpriser.dk har hentet 167 Mb i dag og 304 i går.
Jeg kører i tabel-mode og er givet skyld i en dal af trafikken i går, men i dag har jeg ikke rørt den tab.
Har ingen info om cpu-forbrug.

**Comment by rndfm** [comment]
> Ved ikke om dette er beslægtet eller jeg skal starte nyt issue: Glasswire netmonitor viser at firefox hos elspotpriser.dk har hentet 167 Mb i dag og 304 i går. Jeg kører i tabel-mode og er givet skyld i en dal af trafikken i går, men i dag har jeg ikke rørt den tab. Har ingen info om cpu-forbrug.

Vil det sige at elspotpris.dk siden har hentet så meget data på en dag? Det er i hvert fald ikke meningen.

Jeg kigger ind i dette issue i aften. Så får jeg bekræftet om data forbruget er relateret eller et andet problem.

**Comment by rndfm** [comment]
Jeg har lavet en ny visning som forhåbentlig kan erstatte ApexChart grafen på sigt.

Indtil videre vil jeg gerne høre jeres mening om "Bars" visning. Og om hvad i synes der skal til for at vi kan erstatte den gamle graf visning.

![image](https://user-images.githubusercontent.com/46886958/201544148-8e1f52f3-ffc1-4fda-8376-671751fb3be5.png)

**Comment by hekner** [comment]
Jeg synes ikke der mangler noget - drop endelig den gamle.
Så er vi måske klar til at få optionen med de enkelte lodrette Bars brudt op som næste skridt? (spot-pris, transmission, elafgift og moms?)

**Comment by rndfm** [comment]
Jeg har sat den gamle på pension (Den kan stadig findes) så nu er det den nye som vises som standard.

Og ja. Så er vi tættere på at bryde søjlerne ned så man kan se udregningen.

**Comment by rndfm** [comment]
Den nye visning skulle gerne bruge 0% cpu. Lad mig høre hvis i ser noget andet :)

**Comment by bennybechdk** [comment]
Den nye Søjler/Bars visning virker super godt savner ikke noget fra graf.
Graf havde endda den fejl at prisen mellem 23 og 00 (i sidste døgn) ikke blev vist (glemt at melde det).
Den fejl har den nye ikke.
Så vidt jeg kan se mangler man CO2, men det er ikke noget jeg savner.
